integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMonitoringExporter.java

```diff
@@ -119,24 +130,88 @@ public static void initAll(@Namespaces(2) List<String> namespaces) {
+      isPrometheusDeployed = true;
+    } else {
+      String oldRegex = String.format("regex: %s;%s", domain2Namespace, domain2Uid);
+      String newRegex = String.format("regex: %s;%s", domain1Namespace, domain1Uid);
+      assertDoesNotThrow(() -> editPrometheusCM(oldRegex, newRegex, "istio-system","prometheus"),
+          "Can't modify Prometheus CM, not possible to monitor " + domain1Uid);
+    }
 
     //verify metrics via prometheus
-    String testappPrometheusSearchKey =
-        "wls_servlet_invocation_total_count%7Bapp%3D%22testwebapp%22%7D%5B15s%5D";
-    assertDoesNotThrow(() -> checkMetricsViaPrometheus(testappPrometheusSearchKey,
-        "testwebapp", prometheusPort));
+    String sessionAppPrometheusSearchKey =
+        "wls_servlet_invocation_total_count%7Bapp%3D%22myear%22%7D%5B15s%5D";
+    assertDoesNotThrow(() -> checkMetricsViaPrometheus(sessionAppPrometheusSearchKey, "sessmigr",prometheusPort));
+  }
+
+  /**
+   * Create a domain using model-in-image model with monitoring exporter sidecar.
+   * Add istio configuration with default readinessPort.
+   * Do not add any AdminService under AdminServer configuration.
+   * Deploy istio gateways and virtual service.
+   * Verify server pods are in ready state and services are created.
+   * Verify login to WebLogic console is successful thru istio ingress port.
+   * Deploy Istio provided Prometheus
+   * Verify Weblogic metrics can be processed via istio based prometheus
+   */
+  @Test
+  @DisplayName("Create istio provided prometheus and verify "
+      + "it can monitor Weblogic domain via weblogic exporter sidecar")
+  public void testIstioPrometheusWithSideCar() {
+    // create image with model files
+    logger.info("Create image with model file and verify");
+
+    List<String> appList = new ArrayList();
+    appList.add("sessmigr-app");
+
+    // build the model file list
+    final List<String> modelList = Collections.singletonList(MODEL_DIR + "/model.sessmigr.yaml");
+    String miiImage =
+        createMiiImageAndVerify("miimonexp-istio-image", modelList, appList);
+
+    // docker login and push image to docker registry if necessary
+    dockerLoginAndPushImageToRegistry(miiImage);
+
+    String monitoringExporterSrcDir = Paths.get(RESULTS_ROOT, "monitoringexp", "srcdir").toString();
+    cloneMonitoringExporter(monitoringExporterSrcDir);
+    String exporterImage = assertDoesNotThrow(() -> createPushImage(monitoringExporterSrcDir, "exporter",
+        domain2Namespace, OCIR_SECRET_NAME, getDockerExtraArgs()),
+        "Failed to create image for exporter");
+    String exporterConfig = RESOURCE_DIR + "/exporter/exporter-config.yaml";
+    String managedServerPrefix = domain2Uid + "-managed-server";
+    assertDoesNotThrow(() -> setupIstioModelInImageDomain(miiImage, domain2Namespace, domain2Uid, exporterConfig,
+        exporterImage, managedServerPrefix), "setup for istio based domain failed");
+    if (!isPrometheusDeployed) {
```

Comment:
this section of the code is similar to the method testIstioPrometheusViaExporterWebApp. Maybe you can create a helper method.

Reply:
done